tinygrad/tensor.py

```diff
@@ -133,9 +134,13 @@ def eye(dim, **kwargs): return Tensor([1], **kwargs).slice(((0,dim+1),)).reshape
 
   # TODO: below line, remove use of numpy here and make lazy
   # TODO: requires cumsum to remove numpy
+  
   @staticmethod
-  def arange(stop, start=0, step=1, **kwargs): return Tensor(np.arange(start=start, stop=stop, step=step, dtype=np.float32), **kwargs)
-
+  def cumsum(x: Iterable): return Tensor(itertools.accumulate(x))
```

Comment:
There should be a test for `cumsum` as well.

Reply:
Added test for cumsum, all tests passed.

Reply:
https://github.com/geohot/tinygrad/blob/6738ad3f9e461a00fea08723234f0c07e18b2bac/tinygrad/tensor.py#L137-L138